Simulation Code/Base/Base.ino

```diff
@@ -44,59 +51,59 @@ void checkWaterLevel(int &waterLevelA, int &waterLevelB, int &waterLevelC)
 }
 
 // Function to check and update the status of the pump
-void checkPumpStatus(unsigned long &currentTime, unsigned long &elapsedTime, 
+void checkPumpStatus(WaterPumpFn &waterPump, unsigned long &currentTime, unsigned long &elapsedTime, 
```

Comment:
See comment at the bottom, but the point of a struct is to group similar variables or pin declarations together